### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry/Modules/Tilde): construct the tilde sheaf associated to a module

### PR DESCRIPTION
**Pull request by weihong-xu**
Co-authored-by: Kevin Buzzard <k.buzzard@imperial.ac.uk>
Co-authored-by: Johan Commelin <j.m.commelin@uu.nl>
Co-authored-by: Amelia Livingston <al3717@ic.ac.uk>
Co-authored-by: Sophie Morel <sophie.morel@ens-lyon.fr>
Co-authored-by: Jujian Zhang <jujian.zhang1998@gmail.com>

This PR was done during the AIM workshop Formalizing Algebraic Geometry.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/Quasicoherent/Tilde.lean [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+noncomputable instance (U : (Opens (PrimeSpectrum.Top R))ᵒᵖ) :
+    Module ((𝒪_SpecR).val.obj U) ((Tilde.presheafInAddCommGrp R M).obj U) :=
+  inferInstanceAs $ Module _ (Tilde.sectionsSubmodule R M U)
+
+open Tilde in
+/--
+`M^~` as a sheaf of `𝒪_{Spec R}`-modules
+-/
+noncomputable def TildeInModules : SheafOfModules (𝒪_SpecR) where
```
Comment (joelriou) [inline Mathlib/AlgebraicGeometry/Quasicoherent/Tilde.lean:185]: You should import `AlgebraicGeometry.Modules.Sheaf` and use `Scheme.Modules`. Also, a better location for this file would be `AlgebraicGeometry.Modules.Tilde`.

Comment (jjaassoonn) [reply Mathlib/AlgebraicGeometry/Quasicoherent/Tilde.lean]: I wonder if it is better to make "tilde" inside namespace `ModuleCat`, so we can use dot notation. What do others think?

Comment (joelriou) [reply Mathlib/AlgebraicGeometry/Quasicoherent/Tilde.lean]: `ModuleCat.tilde` seems good to me.

**Comment by jjaassoonn** [comment]
I have put everything inside `ModuleCat` namespace, so we can have `M.tilde`. Not sure if this is the way to go though

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**